### DISCUSSION TITLE
Keep config setting for site enabled, whether primary or not

### DISCUSSION THREAD

**Discussion by jurriaanr** [craftcms/cms#13651] [Ideas] [answered]
I have a code base that is used for multiple projects. The sites for all projects are configured and they are enabled/disabled based on environmental variables. However, the primary site can be different per project.

Right now, when you turn on a site as primary, the enabled status will automatically be true. Fair enough for the site always being enabled. However in doing so, the config yaml file is also changed for the value to be true, instead of the originally configured environmental parameter. Can it not be so that the outcome is that the primary site is always enabled, no matter what the config has as a value, but that the original value is kept. The code should just always return true for the primary site.

(for that matter, having the primary site be configurable through the environment would be an additional plus)

**Comment by brandonkelly** [comment, marked as answer]
It would be pretty confusing to have the Enabled input be editable, even though its value will always be ignored.

> (for that matter, having the primary site be configurable through the environment would be an additional plus)

We can’t do that because changing the primary site has major side effects: all element types which aren’t localizable (addresses, users, etc.) will get stored for the primary site. So when it changes, all of those elements’ content must be reassigned to the new primary site. We wouldn’t be able to catch that if the primary site was just determined by an environment variable, with no easy way to detect changes. (And even if we could, that major side effect of merely changing an environment variable would likely cause a lot of frustration.)

**Comment by jurriaanr** [comment]
I see what you mean by it being confusing, however a little text above just as is the case now goes a long way
